### ISSUE TITLE
[lsp-server] 🐞 incorrect bundling of types for `graphql-language-server`

### ISSUE THREAD

**Issue by phryneas** [graphql/graphiql#3790]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currently, `graphql-language-server@5.3.0` causes type errors with `graphql@16.9.0`. Apparently, this package bundles the types of the development dependency `graphql` into the published bundle, which will then cause TypeScript errors.

Also see the discussion in https://github.com/graphql/graphql-js/issues/4201

### Expected Behavior

As `16.9.0` is a valid `peerDependency`, it should not result in TS warnings.

### Steps To Reproduce

Install  `graphql-language-server@5.3.0` and `graphql@16.9.0`, run `tsc`.

### Environment

See this PR: https://github.com/apollographql/vscode-graphql/pull/175 with a [failing CI job](
https://app.circleci.com/pipelines/github/apollographql/vscode-graphql/812/workflows/7ef38837-7497-42de-aa37-f170622a0ff6/jobs/3925)

**Comment by yaacovCR** [comment]
This would be fixed by the latest release: https://github.com/graphql/graphiql/pull/3763

**Comment by phryneas** [comment]
I don't think that a "move back to 16" is necessarily required to fix this. (I think that's what you're hinting at, right?)
 
The problem is that the `graphql` types end up in the `graphql-language-server` - not as an imported dependency, but a rolled-up type file.

**Comment by phryneas** [comment]
Or no, looking at the source code, I think this might actually not be the case and @JoviDeCroock might have overinterpreted things. The problem is not necessarily the bundling itself, but this statement:

```ts
export const RuleKinds = {
  ...Kind,
  ...AdditionalRuleKinds,
};
```

The spread pulls in all the information about `Kind`.

**Comment by phryneas** [comment]
Yeah, I meant it's not a problem on the build tooling side, which I had assumed you meant.

I think this could fix it, I'm trying it out as we speak:
```ts
export const RuleKinds: Omit<typeof Kind, keyof _AdditionalRuleKinds> &
  _AdditionalRuleKinds = {
  ...Kind,
  ...AdditionalRuleKinds,
};
```

**Comment by phryneas** [comment]
Oh, even simpler...

```ts
export const RuleKinds: _RuleKinds = {
  ...Kind,
  ...AdditionalRuleKinds,
};
```

**Comment by yaacovCR** [comment]
I believe we wanted to move back to v16 as the dev dependency because it’s actually re-exported in one of the packages in total, and when we upgraded, we did it under the assumption that it would not affect users at all.

So we are definitely moving back to v16, which can be released whenever appropriate. @acao please feel free to jump in if I have that wrong.

More broadly, things could be rearchitectured of course!

**Comment by yaacovCR** [comment]
graphiql will still support v17, but it won’t be the version in the dev dependency and I assume will fix this problem.

**Comment by abhishek71994** [comment]
Hello, is this fixed? As I am running into similar CI issues when I upgrade to

```
"graphiql": "^3.8.3"
```

At the end, it gives me this issue
```
node_modules/graphql-language-service/dist/parser/types.d.ts:116:37 - error TS2694: Namespace 'Kind' has no exported member 'LIST_NULLABILITY_OPERATOR'.

116     LIST_NULLABILITY_OPERATOR: Kind.LIST_NULLABILITY_OPERATOR;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/graphql-language-service/dist/parser/types.d.ts:117:30 - error TS2694: Namespace 'Kind' has no exported member 'NON_NULL_ASSERTION'.

117     NON_NULL_ASSERTION: Kind.NON_NULL_ASSERTION;
                                 ~~~~~~~~~~~~~~~~~~

node_modules/graphql-language-service/dist/parser/types.d.ts:118:26 - error TS2694: Namespace 'Kind' has no exported member 'ERROR_BOUNDARY'.

118     ERROR_BOUNDARY: Kind.ERROR_BOUNDARY;
                             ~~~~~~~~~~~~~~


Found 3 errors in the same file, starting at: node_modules/graphql-language-service/dist/parser/types.d.ts:116
```

Looks identical to https://github.com/apollographql/vscode-graphql/pull/175/checks

**Comment by dimaMachina** [comment]
I believe this issue was already fixed, feel free to reply if you still have this issue and I'll reopen this issue!
